lib/runAll.js

```diff
@@ -67,47 +48,32 @@ const runAll = async (
     shell = false,
     stash = true,
     verbose = false,
+    vcsAdapter = undefined,
   },
   logger = console
 ) => {
   debugLog('Running all linter scripts')
 
   const ctx = getInitialState({ quiet })
 
-  const { gitDir, gitConfigDir } = await resolveGitRepo(cwd)
-  if (!gitDir) {
-    if (!quiet) ctx.output.push(NOT_GIT_REPO)
-    ctx.errors.add(GitRepoError)
-    throw createError(ctx)
-  }
+  const adapter = createVcsAdapter(vcsAdapter, { allowEmpty, stash }, logger)
 
-  // Test whether we have any commits or not.
-  // Stashing must be disabled with no initial commit.
-  const hasInitialCommit = await execGit(['log', '-1'], { cwd: gitDir })
-    .then(() => true)
-    .catch(() => false)
+  const { baseDir, shouldBackup } = await adapter.init(ctx, cwd)
```

Comment:
Looking at this and line 59 above, `backup` and `stash` are used interchangeably; maybe we could talk only about backups.

Also, since we already have the constructor step, maybe `ctx` and `cwd` can be passed in there along with  the other options.

Reply:
>Also, since we already have the constructor step, maybe ctx and cwd can be passed in there along with the other options.

moved cwd to contructor. didn't move ctx, because it's modified externally in runAll.

>Looking at this and line 59 above, backup and stash are used interchangeably; maybe we could talk only about backups.

@iiroj  can you please explain - did you mean to replace the option to disable backups (`stash` →  `backup`)?